covid/transform_utils.py

```diff
@@ -126,45 +206,81 @@ def generate_lags(df, column, num_lags=121, lag_timedelta=datetime.timedelta(day
     column_names.extend(lag_column_names)
 
     lags_df = pd.DataFrame(
-        index=pd.Index(data=states, name=STATE_FIELD), columns=column_names
+        index=pd.Index(data=locations, name=by_column), columns=column_names
     )
 
     today = pd.to_datetime(df[DATE_SOURCE_FIELD]).max()
     lags_df[DATE_SOURCE_FIELD] = today
-    for state in states:
+    for location in locations:
         # Start each state looking up today.
         date_to_lookup = today
 
         for lag in range(num_lags):
-            print(f"For field {column}, processing {state} for lag {lag}.")
+            print(f"For field {column}, processing {location} for lag {lag}.")
             # Lookup the historical entry.
             value = df.loc[
-                (df[STATE_FIELD] == state) & (df[DATE_SOURCE_FIELD] == date_to_lookup),
+                (df[by_column] == location) & (df[DATE_SOURCE_FIELD] == date_to_lookup),
                 column,
             ]
 
             if len(value) > 1:
                 raise ValueError("Too many or too few values returned.")
             elif len(value) == 1:
                 value = value.iloc[0]
-                lags_df.loc[state, column_name_formatter.format(lag)] = value
+                lags_df.loc[location, column_name_formatter.format(lag)] = value
 
             date_to_lookup = date_to_lookup - lag_timedelta
 
     lags_df = lags_df.reset_index()
     return lags_df
 
 
-def calculate_state_summary(transformed_df, columns):
+def compute_lagged_frame(
+    frame, num_periods, freq=None, suffix=None, suffix_level=None, subset=None
+):
+    """Computes a data frame with columns that contain the lagged values for the existing columns.
+
+    Args:
+        frame (pandas.DataFrame): data frame to compute the lagged values for
+        num_periods (list of int): number of periods that each new feature column should be lagged by
+        freq (str, optional): time frequency to use in the shift (e.g. 'H' for hourly)
+        suffix (str, optional): suffix to use for titles of the new feature columns
+        suffix_level (int, optional): if columns are MultiIndex, which level to add the suffix to
+        subset (list of str, optional): subset of columns to compute lagged values for
+
+    Returns:
+        lagged_frame (pandas.DataFrame): data frame with the lagged values
+    """
+    if subset is None:
+        subset = frame.columns.tolist()
+
+    if suffix_level is None or suffix_level == len(frame.columns.levels):
+        suffix_level = -1
+
+    lagged_frame = pd.concat(
+        [
+            _reindex_and_rename_columns(
+                frame=frame[subset].shift(periods, freq=freq),
+                index=frame.index,
+                columns=_create_period_column_names(
+                    subset, suffix, periods, suffix_level, frame.columns
```

Comment:
Nit: on the Crystal style guide we require naming arguments, and this feels like a good use case for it: it might be a little easier to read as `_create_period_column_names(columns=subset, suffix=suffix, ...`